crates/sui-json/src/lib.rs

```diff
@@ -294,7 +294,7 @@ fn is_homogeneous_rec(curr_q: &mut VecDeque<&JsonValue>) -> bool {
         // Okay to unwrap since we know values exist
         let curr = match curr_q.pop_front().unwrap() {
             JsonValue::Bool(_) => ValidJsonType::Bool,
-            JsonValue::Number(_) => ValidJsonType::Number,
+            JsonValue::Number(x) => if x.is_u64() { ValidJsonType::Number } else { return false },
```

Comment:
So this check failing technically doesn't imply that the array is not homogenous. it just says the array contains bad data. It could be an array of floats.
We probably want to change the function to return `anyhow::Error`, and return error if this check fails.
But then again I'm being pedantic here. If you need to get this in quicker, we can stamp it.